### PR TITLE
CodeMirror implementation of GraphNodeComment

### LINKED ISSUE #11598: Eliminate lexical dependency

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

CodeMirror-based implementation of GraphNodeComment; removes Lexical dependency.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/gui/src/project-view/components/MarkdownEditor/__tests__/markdown.test.ts [hunk 1]
```diff
@@ -64,6 +64,15 @@ test.each([
...
   },
+  {
+    markdown: '[Link text](<https://www.example.com/index.html>)',
+    expectedLinks: [
+      {
+        text: 'Link text',
+        href: 'https://www.example.com/index.html',
+      },
+    ],
+  },
```
Comment (farmaazon) [inline app/gui/src/project-view/components/MarkdownEditor/__tests__/markdown.test.ts:75]: You may consider adding cases for:
1. Link with spaces and <>
2. Link with spaces without <> (invalid case)
Or we may assume that those cases in image are sufficient. But then I would consider moving them here, as "link" seems to me more basic that images.

File: app/gui/src/project-view/util/__tests__/link.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ],
+  emails: [
+    'example@gmail.com',
+    'EXAMPLE@GMAIL.COM',
+    'example..+hello.world@gmail.com',
+    'a@b.bla',
+    '(a@b.cd)',
+  ],
+  neither: [
+    'https://💩.la/',
```
Comment (farmaazon) [inline app/gui/src/project-view/util/__tests__/link.test.ts:22]: :laughing:

**Review by farmaazon — APPROVED** [review]

File: app/gui/src/dashboard/layouts/AssetDocs/AssetDocs.tsx [hunk 3]
```diff
@@ -48,7 +49,7 @@ export function AssetDocsContent(props: AssetDocsContentProps) {
 
       for (const statement of module.statements()) {
         if (statement instanceof ast.MutableFunctionDef && statement.name.code() === 'main') {
-          return statement.documentationText() ?? ''
+          return normalizedMarkdownToStandard(statement.mutableDocumentationMarkdown().toJSON())
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/layouts/AssetDocs/AssetDocs.tsx:52]: why we turn this into a json? We expect a md string to parse it and display

Comment (kazcw) [reply app/gui/src/dashboard/layouts/AssetDocs/AssetDocs.tsx:52]: The `toJSON` method of `Y.Text` returns the unformatted text content as a `string`.

**Pull request merged**